src/array/kernel_decl.h

```diff
@@ -92,6 +93,32 @@ void BackwardSegmentCmp(NDArray feat,
                         NDArray arg,
                         NDArray out);
 
+/*!
+ * \brief Sparse-sparse matrix multiplication
+ *
+ * \note B is transposed (i.e. in CSC format).
+ */
+template <int XPU, typename IdType, typename DType>
+std::pair<CSRMatrix, NDArray> CSRMM(
+    const CSRMatrix& A,
+    NDArray A_weights,
+    const CSRMatrix& B,
+    NDArray B_weights);
+
+/*!
+ * \brief Sparse-sparse matrix summation.
+ */
+template <int XPU, typename IdType, typename DType>
+std::pair<CSRMatrix, NDArray> CSRSum(
```

Comment:
I think `CSRAdd` is more appropriate.

Reply:
I think "Add" implies a binary operation while "Sum" can be n-ary.  This is an n-ary operation for more efficient gradient computation.

Reply:
Got it.